File: Sponsors/mw-streams-java/src/main/java/com/mwam/hackathon/sinks/ConsoleSink.java
package com.mwam.hackathon.sinks;

/**
 * Prints input to the console.
 */
public class ConsoleSink<T> extends Sink<T> {
    private final String prefix;

    public ConsoleSink() {
        super();
        this.prefix = "";
    }

    public ConsoleSink(String prefix) {
        super();
        this.prefix = prefix;
    }

    @Override
    public void consume(T input) {
        System.out.println(prefix + input);
    }
}

File: Sponsors/mw_streams_python/mws/sinks/console_sink.py
import sys

from typing import TypeVar

from mws.operators.operator import Sink


T = TypeVar("T")


class ConsoleSink(Sink[T]):
    """
    A sink that prints all input events to the console.
    """

    def __init__(self, prefix: str = "") -> None:
        super().__init__()
        self.prefix = prefix

    def consume(self, event: T) -> None:
        print(self.prefix + str(event))
